### PR TITLE
Add `flow` field to `Msg`

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/msgs/models.py [hunk 1]
```diff
@@ -369,7 +369,8 @@ class Msg(models.Model):
     channel = models.ForeignKey(Channel, on_delete=models.PROTECT, null=True, related_name="msgs")
     contact = models.ForeignKey(Contact, on_delete=models.PROTECT, related_name="msgs", db_index=False)
     contact_urn = models.ForeignKey(ContactURN, on_delete=models.PROTECT, null=True, related_name="msgs")
-    broadcast = models.ForeignKey(Broadcast, on_delete=models.PROTECT, null=True, blank=True, related_name="msgs")
+    broadcast = models.ForeignKey(Broadcast, on_delete=models.PROTECT, null=True, related_name="msgs")
```
Comment (rowanseymour) [inline temba/msgs/models.py:372]: just removing `blank=True` which doesn't make sense

File: temba/msgs/models.py [hunk 2]
```diff
@@ -369,7 +369,8 @@ class Msg(models.Model):
...
+    flow = models.ForeignKey("flows.Flow", on_delete=models.PROTECT, null=True, db_index=False)
```
[the remark this answers is not in the document]
Comment (norkans7) [reply temba/msgs/models.py:373]: I guess we want `on_delete=models.SET_NULL`  instead of PROTECT

[the remark this answers is not in the document]
Comment (rowanseymour) [reply temba/msgs/models.py:373]: thought about that but is there ever a situation where we delete a flow before messages?

[the remark this answers is not in the document]
Comment (norkans7) [reply temba/msgs/models.py:373]: Right, I see we release a flow only use delete in workspace release where we first release all messages

**Review by norkans7 — APPROVED** [review]

**Review by ericnewcomer — APPROVED** [review]

**Pull request merged**